ci/envoy-sync-receive.sh

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env bash
+#
+# Copyright Red Hat
+#
+# This script is invoked from .github/workflow/envoy-sync-*.yaml workflows.
+#
+# It merges the specified branch from the upstream envoyproxy/envoy repository
+# into the current branch in the current working directory.  It is assumed that
+# the calling workflow has already checked out the destination repository and
+# switched to the destination branch, in the current working directory before
+# invoking us.
+#
+# - If the merge is successful, it:
+#  - pushes the feature branch to the repository
+#  - creates the associated pull request if it doesn't already exist
+#  - closes the associated issue if it already exists
+#
+# -If the merge is unsuccessful, it:
+#  - leaves the associated pull request untouched if it already exists
+#  - creates the associated issue issue if it doesn't already exist
+#  - adds a comment on the associated issue to describe the merge fail
+#
+
+set -xeuo pipefail
```

Comment:
nit: We are not leaking any credentials with `-x`, are we? For extra caution, I'd remove it. We can always put it back locally for debugging.